### ISSUE TITLE
using segmentations as masks for intensity analysis

### ISSUE THREAD

**Issue by martinschorb** [mobie/mobie-viewer-fiji#355]
we just realized that this is a much more general functionality that we want inside MoBIE:

https://github.com/mobie/yeast-clem-datasets/issues/9

Basically, any segmentation source (or selection of labels) should be available as masks for subsequent analysis of the other sources at the same location.
The basic application would be the intensity quantification inside the mask. This would then be shown as an additional column in the annotations table.

**Comment by constantinpape** [comment]
I agree that this will be super useful to have - but it opens a bit of pandora's box of how much downstream analysis we want to implement inside of MoBIE itself.

**Comment by tischi** [comment]
> but it opens a bit of pandora's box of how much downstream analysis we want to implement inside of MoBIE itself.

I think measuring the intensities in the image segments could be nice as we anyway need this for the PROSPR gene expression per cell measurements. But indeed, I would then try to leave it at that....

But you are right! It does open a very big box and we have to prioritize...

**Comment by martinschorb** [comment]
In my opinion, basic intensity quantification (average, min/max, stdev) inside the masks and displaying the direct result in the annotations table would be the obvious things that MoBIE should do itself.

Anything more fancy could then (at some later point)be implemented by enabling Fiji "macros/scripts" (of any language) acting on the subset of data.

**Comment by constantinpape** [comment]
> In my opinion, basic intensity quantification (average, min/max, stdev) inside the masks and displaying the direct result in the annotations table would be the obvious things that MoBIE should do itself.

Yes, I agree that we should do it because this one is obvious, just make clear that downstream analysis is not the main target application, so that one shouldn't expect more niche features.

> Anything more fancy could then (at some later point)be implemented by enabling Fiji "macros/scripts" (of any language) acting on the subset of data.

Yeah, I guess that would indeed be ideal, but not sure how easy/hard it is on the implementation side.

**Comment by tischi** [comment]
it is hard as one would have to develop image analysis in real-space rather than on a voxel grid.
this is something I really would love to see in "ImageJ2" but I am not sure anyone is actively pursuing this.

**Comment by martinschorb** [comment]
I think this is going into the same direction.

https://forum.image.sc/t/napari-pyclesperanto-assistant-image-array-formats/55156/5

I don't think they can deal with different voxel-sized sources yet. But here @NicoKiaru's  re-sampling of sub-regions might be one approach. Maybe deriving the resulting virtual "voxel size" from the current viewer's pixel size would be the way to go...

**Comment by tischi** [comment]
> Maybe deriving the resulting virtual "voxel size" from the current viewer's pixel size would be the way to go...

I think @petebankhead from QuPath is doing something along those lines where the analysis is done on the voxels of currently visible resolution level, isn't it?
